src/PanelTraits/Create.php

```diff
@@ -112,10 +112,12 @@ public function syncPivot($model, $data, $form = 'create')
 
             if (isset($field['morph']) && $field['morph']) {
                 $values = isset($data[$field['name']]) ? $data[$field['name']] : [];
-                if ($model->{$field['name']}) {
-                    $model->{$field['name']}()->update($values);
-                } else {
-                    $model->{$field['name']}()->create($values);
+
+                foreach ($values as $value) {
+                    $class = $field['model'];
+                    $class = new $class();
+                    $ref = $class->find($value);
```

Comment:
What if that turns out to be `null`? i.e. for some reason `$value` isn't there?

Reply:
check this https://github.com/gotrecillo/CRUD/commit/ea927d3f2b2325b932a80c4bed2633b784519591

Solves the ``` $value = null``` you described + allows for updating